collect_app/src/main/java/org/odk/collect/android/google/SheetsHelper.java

```diff
@@ -0,0 +1,146 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.odk.collect.android.google;
+
+
+import com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.model.BatchUpdateSpreadsheetRequest;
+import com.google.api.services.sheets.v4.model.GridProperties;
+import com.google.api.services.sheets.v4.model.Request;
+import com.google.api.services.sheets.v4.model.SheetProperties;
+import com.google.api.services.sheets.v4.model.Spreadsheet;
+import com.google.api.services.sheets.v4.model.SpreadsheetProperties;
+import com.google.api.services.sheets.v4.model.UpdateSheetPropertiesRequest;
+import com.google.api.services.sheets.v4.model.UpdateSpreadsheetPropertiesRequest;
+import com.google.api.services.sheets.v4.model.ValueRange;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class SheetsHelper {
+    private Sheets sheets;
+
+    SheetsHelper(GoogleAccountCredential credential, HttpTransport transport, JsonFactory jsonFactory) {
+        // Initialize sheets service
+        sheets = new Sheets.Builder(transport, jsonFactory, credential)
+                .setApplicationName("ODK-Collect")
+                .build();
+    }
+
+    public void resizeSpreadSheet(String spreadsheetId, int sheetId, int columnSize) throws IOException {
+
+        // create grid properties with the new column size
+        GridProperties gridProperties = new GridProperties().setColumnCount(columnSize);
+
+        // create sheet properties for the first sheet in the spreadsheet
+        SheetProperties sheetProperties = new SheetProperties()
+                .setSheetId(sheetId)
+                .setGridProperties(gridProperties);
+
+        // Updates properties of the sheet with the specified sheetId
+        UpdateSheetPropertiesRequest updateSheetPropertyRequest = new UpdateSheetPropertiesRequest()
+                .setProperties(sheetProperties)
+                .setFields("gridProperties.columnCount");
+
+        // generate request
+        List<Request> requests = new ArrayList<>();
+        requests.add(new Request().setUpdateSheetProperties(updateSheetPropertyRequest));
+
+        // send the API request
+        sheets.spreadsheets()
+                .batchUpdate(
+                        spreadsheetId,
+                        new BatchUpdateSpreadsheetRequest().setRequests(requests)
+                ).execute();
+    }
+
+    /**
+     * Inserts a new row in the given sheet of the spreadsheet
+     */
+    public void insertRow(String spreadsheetId, String sheetName, ValueRange row) throws IOException {
```

Comment:
@shobhitagarwal1612 this is a good example of the type of method I mention in my overall comment.